core/restore.php

```diff
@@ -0,0 +1,418 @@
+<?php
+/**
+ * @file
+ * Administrative page for restoring backup database and configuration files.
+ *
+ * Point your browser to "http://www.example.com/core/restore.php" and follow
+ * the instructions.
+ *
+ * If you are not logged in using either the site maintenance account or an
+ * account with the "Restore site backups" permission, you will need to
+ * modify the access check statement inside your settings.php file. After
+ * finishing the upgrade, be sure to open settings.php again, and change it
+ * back to its original state!
+ */
+
+/**
+ * Defines the root directory of the Backdrop installation.
+ */
+define('BACKDROP_ROOT', dirname(dirname($_SERVER['SCRIPT_FILENAME'])));
+
+// Change the directory to the Backdrop root.
+chdir(BACKDROP_ROOT);
+
+/**
+ * Global flag indicating that restore.php is being run.
+ *
+ * When this flag is set, various operations do not take place, such as invoking
+ * hook_init() and hook_exit(), css/js preprocessing, and translation.
+ */
+const MAINTENANCE_MODE = 'restore';
+
+/**
+ * Provides an overview of the backup restoration process.
+ *
+ * This page provides cautionary statements before proceeding with a restore.
+ *
+ * @return string
+ *   Rendered HTML form.
+ */
+function restore_info_page() {
+  restore_task_list('info');
+  backdrop_set_title('Restore site backup');
+  $token = settings_get('restore_free_access') ? '1' : backdrop_get_token('restore');
+  $output = '<p>' . st('Use this utility to restore your site\'s database and configuration to a previous version.') . '</p>';
+  $output .= '<p>' . st('For more detailed information, see the Backdrop CMS <a href="!url">documentation on restoring backups</a>.', array(
+    '!url' => 'https://docs.backdropcms.org/documentation/restoring-backups',
+  )) . '</p>';
+
+  $form_action = check_url(backdrop_current_script_url(array(
+    'op' => 'select',
+    'token' => $token,
+  )));
+  $output .= '<form method="post" action="' . $form_action . '">
+    <div class="form-actions">
+      <input type="submit" value="Continue" class="form-submit button-primary" />
+      <a href="' . base_path() . '">Cancel</a>
+    </div>
+  </form>';
+  $output .= "\n";
+  return $output;
+}
+
+/**
+ * Provides a form to select a backup to restore.
+ *
+ * @return string
+ *   Rendered HTML form.
+ */
+function restore_select_page() {
+  restore_task_list('select');
+  backdrop_set_title('Select to backup restore');
```

Comment:
That's odd wording... wouldn't something like "Select a backup to restore from" be a better fit? :wink:

Reply:
Thanks, should just be "Select backup to restore"

Reply:
> should just be "Select backup to restore"

Yes, better. But it seems like you forgot to commit it - it's still the old string here. :grin:

Reply:
It's fixed now!